api/config/v1alpha1/shared_types.go

```diff
@@ -56,6 +58,11 @@ type KubernetesDeploymentSpec struct {
 	// +optional
 	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
 
+	// Image specifies the image used by the container.
+	//
+	// +optional
+	Image *string `json:"image,omitempty"`
```

Comment:
what's the different between a nil point and empty string?

Reply:
pointer since its optional,